### PR TITLE
Issue 233: Fixing pipeline expressions for objects produce strange results in the UI wizard

### PR DESCRIPTION
**Pull request by TrevinTeacutter**

### CODE REVIEW

**Review by willgorman — APPROVED** [review]

File: app/scripts/modules/dcos/job/general.component.html [hunk 1]
```diff
@@ -30,7 +30,7 @@
       Memory (MiB)
     </div>
     <div class="col-md-2">
-      <input type="number" class="form-control input-sm" name="mem" ng-model="$ctrl.general.mem" min="1" placeholder="1"/>
+      <input type="number" class="form-control input-sm" name="mem" ng-model="$ctrl.general.mem" min="1" placeholder="128"/>
```
Comment (mtweten) [inline app/scripts/modules/dcos/job/general.component.html]: Does placeholder actually set a value on the model? I was under the impression that placeholder is meant as a "visual example" and won't actually bind to the ctrl.general.* ng-model. In create server group, for example, we set defaults on the model in the [CommandBuilder](https://github.com/cerner/deck/blob/integration/app/scripts/modules/dcos/serverGroup/configure/CommandBuilder.js#L107-L149). 

Similarly, it looks like we could set the defaults on this "general" model [here](https://github.com/cerner/deck/blob/integration/app/scripts/modules/dcos/job/general.component.js#L13).

File: app/scripts/modules/dcos/serverGroup/configure/wizard/healthChecks.controller.js [hunk 2]
```diff
@@ -13,9 +13,11 @@ module.exports = angular.module('spinnaker.serverGroup.configure.dcos.healthChec
     this.healthCheckProtocols = [HTTP_PROTOCOL, COMMAND_PROTOCOL, TCP_PROTOCOL];
     this.healthCheckPortTypes = ['Port Index', 'Port Number'];
 
-    $scope.command.healthChecks.forEach((hc) => {
-      hc.portType = hc.port ? this.healthCheckPortTypes[1] : this.healthCheckPortTypes[0];
-    });
+    if (typeof $scope.command.healthChecks !== 'string') {
```
Comment (mtweten) [inline app/scripts/modules/dcos/serverGroup/configure/wizard/healthChecks.controller.js]: Want to use the isHealthChecksValid function  here too?

**Review by mtweten — APPROVED** [review]

**Comment by mtweten** [comment]
+1 after comments addressed

File: app/scripts/modules/dcos/job/general.component.html (hunk 1 above)
Comment (TrevinTeacutter) [reply app/scripts/modules/dcos/job/general.component.html]: For some reason I was under the impression it set that, should be able to fix that issue.

File: app/scripts/modules/dcos/serverGroup/configure/wizard/healthChecks.controller.js (hunk 2 above)
Comment (TrevinTeacutter) [reply app/scripts/modules/dcos/serverGroup/configure/wizard/healthChecks.controller.js]: Must've missed this one.

**Pull request merged**